python/tests/data/gen_date_data.py

```diff
@@ -0,0 +1,26 @@
+"""Generate test data."""
+# TODO: Once we have a writer, replace this script.
+from delta import configure_spark_with_delta_pip
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as F
+
+builder = (
+    SparkSession.builder.appName("MyApp")
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+    .config(
+        "spark.sql.catalog.spark_catalog",
+        "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+    )
+)
+
+spark = configure_spark_with_delta_pip(builder).getOrCreate()
+
+df = spark.range(5).repartition(1).withColumn("date", F.lit("2021-01-01").cast("date"))
+df.write.partitionBy("date").format("delta").save("date_partitioned_df")
+
+df = (
+    spark.range(5)
+    .repartition(1)
+    .withColumn("date", F.lit("2021-01-01").cast("timestamp"))
+)
+df.write.partitionBy("date").format("delta").save("timestamp_partitioned_df")
```

Comment:
Do you think we need to keep this generation file? The generated dataset should be sufficient.

Reply:
I thought it might be helpful, but I can remove it. Once this is merged, I'll rebase #566 and rewrite the tests to use the pyarrow based writer to validate the round trip.

Reply:
Deleted.